front50-azure/src/main/java/com/netflix/spinnaker/front50/model/AzureStorageService.java

```diff
@@ -88,6 +92,11 @@ public boolean supportsVersioning() {
   public <T extends Timestamped> T loadObject(ObjectType objectType, String objectKey) {
     String key = buildKeyPath(objectType.group, objectKey, objectType.defaultMetadataFilename);
     try {
+      /* Issue with special character(space)handling in dinghy file.
+       * https://github.com/spinnaker/spinnaker/issues/6221
+       * Container Blob Reference key generation has the logic of double encode of special chars
+       * Decode the key to avoid/fix double encode of special characters.*/
+      key = URLDecoder.decode(key, "ISO-8859-1");
```

Comment:
This should REALLY not be part of any PR to change azure blob permissions and instead a separate PR to make changes and have discussions around this change.

Reply:
Agreed...

I would be happy to submit a new pull request if @undetiramesh cannot get to it, or minds?